File: euler10.c
/*Correctly solves Project Euler problem 10*/
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

long sum = 2;
long limit = 2000000;

/*Returns 0 if not prime, and 1 if prime*/
int checkPrime(int num) {
   if ( num % 2 == 0 ) {
      return 0;
   }
   for ( int i = 3; i <  sqrt(num)+1; i += 2 ) {
      if ( num % i == 0 ) {
         return 0;
      }
   }
   return 1;
}

int main()
{
   for (long i = 3; i < limit; i+=2) {
      if ( checkPrime(i) == 1 ) {
         printf("Found prime %lu\n",i);
         /*sum += i;*/
      }
   }
   printf("The Sum of all prime numbers less than %lu is %lu\n", limit, sum);
}

File: euler10.rb
#!/usr/bin/ruby
#Project Euler 10

limit = 2_000_000
num = 3
primes = [2]

def checkPrime(num, primes)
  for p in primes.select { |x| x <= Math.sqrt(num) } do
    if num % p == 0 then
      return false
    end
  end
  return true
end

while num < limit do
   if checkPrime(num, primes) then
      primes << num
   end
   num += 2
end

sum = primes.inject { |result, element| result + element }
puts "The sum of all primes less than #{limit} is #{sum}"
